docs/references/message-execution-properties.mdx

```diff
@@ -2,67 +2,55 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
-
-To illustrate this, let's consider the above example code again, and assume the method example is called twice in parallel, the resulting calls being Call 1 and Call 2. The following illustration shows two possible message orderings. On the left, both messages of the first call are scheduled first, and only then the second call's messages are executed (the dotted circles are used to denote the callback messages). On the right, you can see another possible message scheduling, where the first messages of each call are executed first and then their callback messages. 
-
-<div class="text--center">
-<img src="/img/docs/interleaving_calls.png" alt="Interleaving Calls" width="500"/>
-</div>
-
-
-### Property 5
-
-On a trap, modifications to the canister's state for the current message are not applied.
+Property 3 provides a guarantee on the execution order of messages on a target canister. However, if multiple calls interleave, one cannot assume additional ordering guarantees for these interleaving calls. To illustrate this, let's consider the above example code again, and assume the method `example` is called twice in parallel, the resulting calls being Call 1 and Call 2. The following illustration shows two possible message orderings. On the left, the first call's messages are scheduled first, and only then the second call's messages are executed. On the right, you can see another possible message scheduling, where the first messages of each call are executed first. Your code should result in a correct state regardless of the message ordering.
 
-For example, if a trap in the second message (dotted circle) of the above example occurs, canister state changes resulting from that message are discarded. However, note that any state changes that happened in the first message (solid circle) have been applied if that message executed successfully.
+![example_orderings](./_attachments/example_orderings.png)
 
+- **Property 5**: On a trap or panic, modifications to the canister state for the current message are not applied.
 
-### Property 6
+For example, if a trap in the second message (blue box) of the above example occurs, canister state changes resulting from that message, even earlier in the blue box, are discarded. However, note that any state changes from earlier messages and in particular the first message (orange box) have been applied, as that message executed successfully.
 
-Inter-canister calls are not guaranteed to make it to the destination canister. If the call does reach the destination canister, the destination canister can trap or return a **reject** response while processing the call.
+- **Property 6**: Inter-canister calls are not guaranteed to make it to the destination canister, and if a call does reach the destination canister, the destination canister can trap or return a reject response while processing the call.
```

Comment:
Should we add that the destination canister can simply NOT reply?

```suggestion
- **Property 6**: Inter-canister calls are not guaranteed to make it to the destination canister, and if a call does reach the destination canister, the destination canister can trap or return a reject response while processing the call. It could also choose to never produce a response.
```

Reply:
I wouldn't say that it can choose not to reply; if it does not reply, the system will generate a response for it (a canister error). But it can delay producing the response for arbitrarily long, potentially infinitely long if it's willing to put the cycles in. We can (and IMO should) say that.

Reply:
Agreed, I explain this in the paragraph below, using some of your wording. Intentionally keeping this vague though.